rules/prefer-event-target.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const {matches} = require('./selectors/index.js');
+
+const MESSAGE_ID = 'prefer-event-target';
+const messages = {
+	[MESSAGE_ID]: 'Prefer `EventTarget` over `EventEmitter`.',
+};
+
+const eventEmitterSuperClassSelector = [
+	matches(['ClassDeclaration', 'ClassExpression']),
+	'[body.type="ClassBody"]',
+	' > ',
+	'Identifier.superClass[name="EventEmitter"]',
+].join('');
+
+const newEventEmitterSelector = [
+	matches(['NewExpression', 'CallExpression']),
```

Comment:
```suggestion
	'NewExpression',
```

Reply:
I think probably we need the `CallExpression` selector here to match the below codes.

```js
// 1
EventEmitter()

// 2
const emitter = EventEmitter()
```

Reply:
In theory, we should not allow `EventEmitter` at all, in following examples, `EventEmitter` can use in some other plcaes to construct.

```js
// pass `EventEmitter` to another function
foo(EventEmitter);
```

```js
foo.bar = EventEmitter;
```

```js
foo = {bar: EventEmitter};
```

but I assume you only want to check common cases, "super class" and "new expression".

```js
EventEmitter()
```

is not common.